packages/grid/_modules_/grid/components/columnHeaders/ColumnHeaderMenuIcon.tsx

```diff
@@ -16,17 +14,17 @@ export function ColumnHeaderMenuIcon(props: ColumnHeaderFilterIconProps) {
   const { column } = props;
   const apiRef = React.useContext(GridApiContext);
   const columnMenuState = useGridSelector(apiRef, gridColumnMenuStateSelector);
-  const columnMenuButtonId: string = useId();
-  const columnMenuId: string = useId();
+  const columnMenuButtonId = `${column.field}-column-menu-button`;
```

Comment:
This looks nice but the problem I think comes if the field starts with a number, then the id would not be valid.

Reply:
It seems that it would also fail with two data grids with the same column name.

Reply:
> This looks nice but the problem I think comes if the field starts with a number, then the id would not be valid.

HTML5 removed this restriction. The [only](https://www.w3.org/TR/2011/WD-html5-20110525/elements.html#the-id-attribute) requirement left is to have more than one character. I'm open for the format of the calculated id, we could also use the field at the end.

> It seems that it would also fail with two data grids with the same column name.

This requires the alternative path I added in the description. We can create a random id for the column and store it in the state or we create a stable unique prefix for the entire grid and all ids.

Reply:
We can also disable portal in popper and/or render the menu in a container.

What issue is the `Id` solving exactly? In the first implementation, there was no Id.

Reply:
> We can also disable portal in popper and/or render the menu in a container.

I didn't understand. Even without the portal we need to set `aria-labelledby`.

> What issue is the Id solving exactly? In the first implementation, there was no Id.

Currently, the user has to pass an id and aria-labelledby to `apiRef.current.showColumn` or the screen reader won't read the menu label. This PR removes the need to pass these arguments.

Reply:
> What issue is the Id solving exactly? In the first implementation, there was no Id.

@dtassone The first implementation was, more or less saying: "a11y, I don't care". The id allows linking the popup with its action and vice-versa. https://www.w3.org/TR/wai-aria-practices/examples/menu-button/menu-button-links.html

Reply:
> HTML5 removed this restriction.

@m4theushw To some extend, try this command: `document.querySelector('.121')` or `document.querySelector('#121')`, it fails.

Reply:
> To some extent, try this command: document.querySelector('.121') or document.querySelector('#121'), it fails (tested on Chrome and Safari).

@oliviertassinari Good to know. `document.querySelector` requires a valid CSS selector and `#123` is valid in HTML5 but not in CSS. `document.getElementById('123')` doesn't throw any error. Anyway, I'll put the field at the end.

Reply:
For the solution, IMHO, the cleanest solution is to change the structure. I think that each header column icon should have its own column menu. It shouldn't be shared. Compare the UX between https://elastic.github.io/eui/#/tabular-content/data-grid and ours when opening multiple column menu. If we change the structure, then it becomes trivial to share the same id with `useId()`.